src/pyk/proof/reachability.py

```diff
@@ -68,6 +68,72 @@ def summary(self) -> Iterable[str]:
         ]
 
 
+class AGBMCProof(AGProof):
```

Comment:
```suggestion
class BMCProof(Proof):
```

* Consider `BMCProof` as name for conciseness.
* Having `AGProof` as a superclass might not be justified in this case, consider inheriting from `Proof` instead.

Reply:
We'll be renaming these classes soon anyway, so I'll leave it to that: https://github.com/runtimeverification/pyk/issues/355